cmake: Build fails in an environment without `python` binary

**Describe the bug**
The commit 9de5bc91093184bb1a57d031e76c089c29806d23 kills our build with `FATAL_ERROR` because the function `determine_python_version` can't find a binary called `python`.  Python 3 in some distro is named `python3`.  With some configuration, there is no alias to `python` or no Python 2 as `python`.  Thus it `FATAL_ERROR`.

**To Reproduce**
1. remove `python` binary.
2. build.

**Expected behavior**
Successfully build with and only with `python3`

**Impact**
Our CI failed.

**Screenshots or console output**
```
CMake Error at zephyr/cmake/backports/FindPythonInterp.cmake:46 (message):
  Failed to test python version
Call Stack (most recent call first):
  zephyr/cmake/backports/FindPythonInterp.cmake:58 (determine_python_version)
  zephyr/cmake/app/boilerplate.cmake:103 (include)
  CMakeLists.txt:3 (include)
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Native
 - Commit SHA or Version used: 9de5bc91093184bb1a57d031e76c089c29806d23

--- a/cmake/backports/FindPythonInterp.cmake
+++ b/cmake/backports/FindPythonInterp.cmake
@@ -54,7 +54,7 @@ endfunction()
 # and choose it if it does. This gives this executable the highest
 # priority, which is expected behaviour.
 find_program(PYTHON_EXECUTABLE python)
-if(NOT (${PYTHON_EXECUTABLE} STREQUAL FIND_PY-NOTFOUND))
+if(NOT (${PYTHON_EXECUTABLE} STREQUAL PYTHON_EXECUTABLE-NOTFOUND))
   determine_python_version(${PYTHON_EXECUTABLE} ver)
   if(${ver} VERSION_LESS PythonInterp_FIND_VERSION)
 	# We didn't find the correct version on path, so forget about it

